src/Database/Type/GeometryType.php

```diff
@@ -70,6 +77,9 @@ public function toDatabase($value, DriverInterface $driver): ?string
         }
 
         $geometry = Geometry::parse($value)->getGeometry();
+        if ($this->decorator !== null) {
+            $geometry = ($this->decorator)($geometry);
```

Comment:
While it is pretty much the same thing, another option would be to use `call_user_func`:

```suggestion
            $geometry = call_user_func($this->decorator, $geometry);
```